engine/classes/ElggEntity.php

```diff
@@ -1068,6 +1068,47 @@ public function canEdit($user_guid = 0) {
 	}
 
 	/**
+	 * Can a user delete this entity?
+	 *
+	 * @tip Can be overridden by registering for the permissions_check:delete plugin hook.
+	 *
+	 * @param int $user_guid The user GUID, optionally (default: logged in user)
+	 *
+	 * @return bool Whether this entity is deletable by the given user.
+	 * @since 1.11
+	 * @see elgg_set_ignore_access()
+	 */
+	public function canDelete($user_guid = 0) {
+		$user_guid = (int) $user_guid;
+
+		if (!$user_guid) {
+			$user_guid = _elgg_services()->session->getLoggedInUserGuid();
+		}
+
+		// need to ignore access and show hidden entities for potential hidden/disabled users
+		$ia = elgg_set_ignore_access(true);
+		$show_hidden = access_show_hidden_entities(true);
+		
+		$user = _elgg_services()->entityTable->get($user_guid, 'user');
```

Comment:
Well, you are passing the user entity to the hook, so the plugin author will assume it's a fully loaded entity. If then they try to access metadata on a hidden entity, I am not sure it will get loaded. I am not sure about the behavior since services refactor, so perhaps we need to add a test stab for this particular case to make sure it doesn't fail.
1. Create user with some metadata
2. Disable user
3. Create an object owned by user
4. Disable access overrides
5. Disable hidden overrides
6. Add a hook for can delete to check for user metadata
7. Delete an object